rsp_vision/console_application/app.py

```diff
@@ -15,64 +16,152 @@
+        Format: Mouse_Id_Hemisphere_BrainRegion_Monitor_position.\n \
+        Example (1 day): AK_1111739_hL_RSPd_monitor_front\n \
+        Example (2 days): BY_IAA_1117276_hR_RSPg_monitor_front\n \
+        Example (1 day, BIG): AS_1112809_hL_V1_monitor_front-right_low\n \
+        Example (2 days, big file): CX_1112654_hL_RSPd_monitor_front\n \
+        Example (2 days, big file): CX_1112837_hL_RSPd_monitor_front\n \
+        Error: CX_122_2_hR_RSPd_monitor_right\n \
+        📄"
+    )
+    config, swc_blueprint_spec = read_config_and_logging()
+    analysis_pipeline(folder_name, config, swc_blueprint_spec)
+
+
+def get_all_sf_tf_datasets(path: str) -> list:
+    """Returns a list of all the datasets in the folder containing the string
+    "sf_tf" in their name.
 
     Parameters
     ----------
-    func : object
-        function to decorate
+    path : str
+        Path to the folder containing the datasets.
 
     Returns
     -------
-    object
-        decorated function
+    list
+        List of all the datasets in the folder containing the string "sf_tf"
+        in their name.
     """
 
-    def inner_function(*args, **kwargs):
-        try:
-            func(*args, **kwargs)
-        except Exception as e:
-            rich.print("Something went wrong 😱")
-            logging.exception(e)
+    only_sf_tf_files = []
+    for filename in os.listdir(path):
+        if "sf_tf" in filename:
+            filename = filename.split("_sf_tf")[0]
+            only_sf_tf_files.append(filename)
+    only_sf_tf_files.sort()
+    return only_sf_tf_files
+
+
+def cli_entry_point_array(job_id):
+    """This is the entry point for the CLI application when running on the
+    cluster. It takes a job id as input and runs the analysis on the dataset
+    corresponding to that job id. I.e. if the job id is 0, it will run the
+    analysis on the first dataset in the folder containing the string "sf_tf"
+    in its name.
+    It also updates the analysis_success.log file with the results of the
+    analysis. If there is an error, it will be logged in the error column of
+    the analysis_success.log file.
 
-    return inner_function
+    Parameters
+    ----------
+    job_id : _type_
+        Job id of the dataset to analyse.
+    """
+    config, swc_blueprint_spec = read_config_and_logging(
+        is_local=False, job_id=job_id
+    )
 
+    allen_folder = config["paths"]["allen-dff"]
+    only_sf_tf_files = get_all_sf_tf_datasets(allen_folder)
+    dataset = only_sf_tf_files[job_id]
+
+    analysis_success_table = AnalysisSuccessTable(
+        swc_blueprint_spec.local_path
+    )
 
-@exception_handler
-def analysis_pipeline() -> None:
-    """Entry point of the program.
+    logging.info(f"Trying to analyse:{dataset}, job id: {job_id}")
+    try:
+        analysis_pipeline(dataset, config, swc_blueprint_spec)
+        analysis_success_table.update(
+            dataset_name=dataset,
+            date=str(datetime.datetime.now()),
```

Comment:
could the `date=str(datetime.datetime.now()),` and the message be handled at the level of `AnalysisSuccessTable`? for example `analysis_sucess=True` at this level and then all logic on what message to print are centralised on the table itself